### ISSUE TITLE
Add a Game object and rework the game loop

### ISSUE THREAD

**Issue by tomiy** [tomiy/openrider#2]
Right now the entry point is canvas_ride in bootstrap.js, which initializes a Track object and its Bike object and pushes its update method to the instances array with a setInterval that loops through it
The problem is that updating and rendering is done at the same time, so we don't have granularity over fps while keeping the game mechanics the same (25 ups, 40ms ticks)

My proposal is to have a Game object that's the entry point and handles creating a new Track and Bike, and also has a run() method that updates every x tick and renders as often as it can with requestAnimationFrame
Consider adding a constant for the tick rate since it's used in the Ghost* classes

**Comment by tomiy** [comment]
Side note: this could also hold a lot of the shit that's currently in bootstrap.js such as keyboard and mouse events

**Comment by tomiy** [comment]
Class methods would look something like this
```
constructor() {
	this.lastTime = performance.now();
	this.timer = performance.now();
	this.ms = 1000 / 25 /*GAME_FPS*/;
	this.delta = 0;
	this.frames = 0;
	this.updates = 0;
}

run() {
	requestAnimationFrame(() =>run()); // maybe push that to the instances array
	let now = performance.now();
	this.delta += (now - this.lastTime) / this.ms;
	this.lastTime = now;
	while (this.delta >= 1) {
		update(); // where
		this.updates++;
		this.delta--;
	}
	render(); // where
	this.frames++;
	if (performance.now() - this.timer > 1000) {
		this.timer += 1000;
		console.log(this.updates + " ups, " + this.frames + " fps"); // Add debug in the canvas too
		this.updates = 0;
		this.frames = 0;
	}
}
```

**Comment by tomiy** [comment]
Partly done, but now we need to have timescale based updates to get 1) smoother framerates 2) granularity without impacting gameplay

**Comment by tomiy** [comment]
Solved in #13
